src/shims/unix/linux/fd/event.rs

```diff
@@ -59,11 +61,17 @@ impl FileDescriptor for Event {
         bytes: &[u8],
     ) -> InterpResult<'tcx, io::Result<usize>> {
         let v1 = self.val.get();
-        // FIXME handle blocking when addition results in exceeding the max u64 value
-        // or fail with EAGAIN if the file descriptor is nonblocking.
-        let v2 = v1.checked_add(u64::from_be_bytes(bytes.try_into().unwrap())).unwrap();
+        let bytes = bytes
```

Comment:
```suggestion
        let bytes: [u8; 8] = bytes
```
This type inference is a bit too much magic for my taste.